portal.go

```diff
@@ -832,6 +900,11 @@ func (portal *Portal) HandleMatrixMessage(evt *event.Event) {
 		}
 	}
 
+	if time.Now().Sub(time.Unix(evt.Timestamp, 0)).Seconds() > 10 {
```

Comment:
The time comparison also looks wrong since event timestamps are milliseconds

```suggestion
	if time.Since(time.UnixMilli(evt.Timestamp)) > 10 * time.Second {
```